src/index.tsx

```diff
@@ -25,6 +25,55 @@ import { LiveChatLoaderProvider } from 'react-live-chat-loader';
 import { DialogManagerProvider } from 'app/contexts/dialog-manager/ActionDialogManager.context';
 import envService from 'app/core/services/env.service';
 
+const CANONICAL_DRIVE_HOSTNAME = 'drive.internxt.com';
+const CANONICAL_DRIVE_ORIGIN = `https://${CANONICAL_DRIVE_HOSTNAME}`;
+
+const buildSafeCanonicalUrl = () => {
+  const isBrowser = typeof globalThis !== 'undefined';
+
+  if (!isBrowser) {
+    return CANONICAL_DRIVE_ORIGIN;
+  }
+
+  try {
+    const { pathname, search, hash } = globalThis.location;
+    const relativeUrl = `${pathname}${search}${hash}`;
+    const normalizedUrl = new URL(relativeUrl, CANONICAL_DRIVE_ORIGIN);
+
+    if (normalizedUrl.origin !== CANONICAL_DRIVE_ORIGIN) {
+      return CANONICAL_DRIVE_ORIGIN;
+    }
+
+    return normalizedUrl.toString();
+  } catch {
+    return CANONICAL_DRIVE_ORIGIN;
+  }
+};
+
+const enforceCanonicalDriveDomain = () => {
+  const isBrowser = typeof globalThis !== 'undefined';
+
+  if (!isBrowser || !envService.isProduction()) {
+    return;
+  }
+  const hostname = globalThis.location.hostname;
+  const isPreviewDeployment = hostname.includes('.pages.dev');
```

Comment:
Perhaps it is also better for preview environments to enable an environment variable, as this may be different at any given time and may not work correctly. This way, it couples with Cloudflare's preview links.